src/TConsole.h

```diff
@@ -117,12 +117,12 @@ class TConsole : public QWidget
     TConsole* createBuffer(const QString& name);
     void scrollDown(int lines);
     void scrollUp(int lines);
-    void print(const QString&, const QColor fgColor, const QColor bgColor);
+    void print(const QString&, QColor fgColor, QColor bgColor);
```

Comment:
Assuming that `QColor` (and container types like `QMap` etc.) is NOT a *plain old data* type **is** the `const` totally redundant though? Doesn't the `const` make it possible/safe/optimisable to use Qt's Copy-On-Write sematics because we are claiming that the copy that is made of the supplied **value** can still be a shallow one because it is not going to be modified... On the other hand I may have the wrong stick entirely, let alone the wrong end of the right one... 🤷‍♂️

Reply:
It is still `const` in the function definition - it just wanted it gone from the declaration:

https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

https://github.com/llvm-mirror/clang-tools-extra/blob/master/test/clang-tidy/readability-avoid-const-params-in-decls.cpp

Reply:
Okay...